purchase_stock_analysis/model/purchase_stock_analysis.py

```diff
@@ -0,0 +1,181 @@
+    state = fields.Selection(
+        [('draft', 'Request for Quotation'),
+         ('confirmed', 'Waiting Supplier Ack'),
+         ('approved', 'Approved'),
+         ('except_picking', 'Shipping Exception'),
+         ('except_invoice', 'Invoice Exception'),
+         ('done', 'Done'),
+         ('cancel', 'Cancelled')], readonly=True)
+    product_id = fields.Many2one('product.product', readonly=True)
+    picking_type_id = fields.Many2one('stock.warehouse', readonly=True)
+    location_id = fields.Many2one('stock.location', 'Destination',
+                                  readonly=True)
+    partner_id = fields.Many2one('res.partner', 'Supplier', readonly=True)
+    pricelist_id = fields.Many2one('product.pricelist', 'Pricelist',
+                                   readonly=True)
+    date_approve = fields.Date('Date Approved', readonly=True)
+    expected_date = fields.Date('Expected Date', readonly=True)
+    validator = fields.Many2one('res.users', 'Validated By', readonly=True)
+    product_uom = fields.Many2one('product.uom', 'Reference Unit of Measure',
+                                  required=True)
+    company_id = fields.Many2one('res.company', 'Company', readonly=True)
+    user_id = fields.Many2one('res.users', 'Responsible', readonly=True)
+
+    delay = fields.Float(
+        'Days to Validate', digits=(16, 2), readonly=True,
+        help='Days between the Order Date and the Date Approval')
+
+    # was delay_pass in purchase analysis
+    days_to_deliver = fields.Float(
+        digits=(16, 2), readonly=True,
+        help='Days between the Order Date and the initial Scheduled Date')
+
+    days_initial_to_updated_schedule = fields.Float(
+        digits=(16, 2), readonly=True,
+        help='Days from initial to updated Scheduled Date of Delivery')
+
+    days_schedule_to_actual_delivery = fields.Float(
+        digits=(16, 2), readonly=True,
+        help='Days from the updated Scheduled Date of Delivery and the actual'
+        'Delivery Date')
+
+    unit_quantity = fields.Integer('Unit Quantity', readonly=True)
+    price_total = fields.Float('Total Price', readonly=True)
+    price_average = fields.Float('Average Price', readonly=True,
+                                 group_operator="avg")
+    negociation = fields.Float('Purchase-Standard Price', readonly=True,
+                               group_operator="avg")
+    price_standard = fields.Float(readonly=True, group_operator="sum")
+    category_id = fields.Many2one('product.category', readonly=True)
+
+    _order = 'date_order desc, price_total desc'
+
+    def init(self, cr):
+        tools.sql.drop_view_if_exists(cr, 'purchase_stock_analysis')
+        cr.execute("""
+            create or replace view purchase_stock_analysis as (
+                WITH currency_rate (currency_id, rate, date_start, date_end)
+                AS (
+                    SELECT r.currency_id, r.rate, r.name AS date_start,
```

Comment:
I mean this: https://github.com/odoo/odoo/blob/8.0/addons/sale/report/sale_report.py#L128